ui/lesson-renderer/src/main/java/org/cru/godtools/tool/lesson/ui/LessonPageAdapter.kt

```diff
@@ -0,0 +1,48 @@
+package org.cru.godtools.tool.lesson.ui
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.Observer
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import org.ccci.gto.android.common.recyclerview.adapter.SimpleDataBindingAdapter
+import org.cru.godtools.tool.lesson.databinding.LessonPageBinding
+import org.cru.godtools.tool.lesson.ui.controller.LessonPageController
+import org.cru.godtools.tool.lesson.ui.controller.bindController
+import org.cru.godtools.xml.model.Manifest
+
+class LessonPageAdapter @AssistedInject internal constructor(
+    @Assisted lifecycleOwner: LifecycleOwner,
+    private val controllerFactory: LessonPageController.Factory
+) : SimpleDataBindingAdapter<LessonPageBinding>(lifecycleOwner), Observer<Manifest?> {
+    @AssistedFactory
+    interface Factory {
```

Comment:
I haven't seen Factory before but searching the code base we use it in a few spots. What makes it different from a regular ```Class```/```Interface```? Or is it just a design pattern?

Reply:
this is a Dagger feature, take a look at this documentation: https://dagger.dev/dev-guide/assisted-injection.html

I can explain in more detail how this works via a video call later if desired, it's not something that would be easy to capture in github comments

Reply:
`Factory` is just the name I gave the interface, it's not a kotlin keyword and doesn't have any significance.

Reply:
Ah okay, gotcha. The ```@AssistedFactory``` is what threw me off.